app/auth/web2auth.tsx

```diff
@@ -104,12 +149,53 @@ function Login() {
 }
 
 function Signup() {
+  const [email, setEmail] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const { mutateAsync: signup, isPending } = useWeb2Signup();
+
   const [rememberMeChecked, setRememberMeChecked] = useState(false);
 
+  async function onPressSignup() {
+    try {
+      await signup({
+        email: email,
+        username: username,
+        password,
+      });
+    } catch (e) {
+      console.log(e);
+      if (e instanceof AxiosError) {
+        console.log(e.response?.data);
+      }
+
+      // FIXME: report errors to UI
+
+      return;
+    }
+
+    router.dismissAll();
+    router.replace("/");
```

Comment:
Shouldn't you redirect for succesfull signup only?? I see that we check for need of auth in home layout, it's fine if you have scheduled these changes for a later PR.

> // FIXME: *actually* perform a check if user is logged in

> const needsLogin = true;"

Reply:
> Shouldn't you redirect for succesfull signup only??
That is what the `return` statement in `catch` block is for

> I see that we check for need of auth in home layout
That will require knowing if the user is signed-in, which in turn requires a persistent storage to store current auth details.

> it's fine if you have scheduled these changes for a later PR.
Yeah. With the SQLite setup.